src/main/java/com/example/demo/config/BeanConfig.java

```diff
@@ -1,28 +0,0 @@
-package com.example.demo.config;
-
-import org.springframework.beans.factory.annotation.Qualifier;
-import org.springframework.boot.jdbc.DataSourceBuilder;
-import org.springframework.context.annotation.Bean;
-import org.springframework.context.annotation.ComponentScan;
-import org.springframework.context.annotation.Configuration;
-import org.springframework.web.servlet.config.annotation.EnableWebMvc;
-
-import javax.sql.DataSource;
-
-@Configuration
-@EnableWebMvc
-@ComponentScan(basePackages = "com.example")
-public class BeanConfig {
-
-    @Bean
-    @Qualifier("defaultDataSource")
-    public DataSource getDataSource() {
```

Comment:
to jest tworzone na podstawie wpisow w pliku application.properties. nie ma ptrzeby robic to recznie.
to wlasnie zalatwia magia Spring Boota. gdybys potrzebowal czegos naprawde niestandardowego to wteyd mozna tak. a nawet jak juz, to zobacz, ze duplikujesz sobie dane.  w application.properties masz adres do bazy i tutaj w kodzie tez masz. nie duplikujemy takich informacji. Spring pozwala sobie wrzucic dane z application.properties do kodu, wystarczy jako parametr ustawic w tym przypadku klase DataSourceProperties